src/repository/error_repo.rs

```diff
@@ -0,0 +1,83 @@
+use {
+    crate::{
+        crypto::{HashAlgorithm, HashValue},
+        interchange::DataInterchange,
+        metadata::{MetadataPath, MetadataVersion, TargetDescription, TargetPath},
+        repository::{RepositoryProvider, RepositoryStorage},
+        Error, Result,
+    },
+    futures_io::AsyncRead,
+    futures_util::future::{BoxFuture, FutureExt},
+    parking_lot::Mutex,
+    std::sync::Arc,
+};
+
+pub(crate) struct ErrorRepository<R> {
+    repo: R,
+    fail_stores: Arc<Mutex<bool>>,
+}
+
+impl<R> ErrorRepository<R> {
+    pub(crate) fn new(repo: R) -> Self {
+        Self {
+            repo,
+            fail_stores: Arc::new(Mutex::new(false)),
+        }
+    }
+
+    pub(crate) fn fail_stores(&self, fail_stores: bool) {
```

Comment:
this method name implies it will fail all stores, but the RepositoryStorage impl only fails metadata stores.  A more specific method name or an impl that also fails target stores would be more clear to me.